x-pack/plugins/infra/server/lib/alerting/common/utils.ts

```diff
@@ -210,3 +220,49 @@ export const shouldTermsAggOnContainer = (groupBy: string | string[] | undefined
     ? groupBy.includes(KUBERNETES_POD_UID)
     : groupBy === KUBERNETES_POD_UID;
 };
+
+export const fetchAlertbyAlertUUID = async (
+  ruleDataClient: PublicContract<IRuleDataClient>,
+  alertUuid: string
+) => {
+  const request = {
+    body: {
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                [ALERT_UUID]: alertUuid,
+              },
+            },
+          ],
+        },
+      },
+      size: 1,
+    },
+    allow_no_indices: true,
+  };
+  const { hits } = await ruleDataClient.getReader().search(request);
+  return hits?.hits;
+};
+
+// fetch alert context from Alerts-As-Data
+export const getContextForRecoveredAlerts = async (
+  ruleDataClient: PublicContract<IRuleDataClient>,
+  alertUuid: string | null
+): Promise<{ [x: string]: any }> => {
+  const alertHits = alertUuid ? await fetchAlertbyAlertUUID(ruleDataClient, alertUuid) : undefined;
```

Comment:
If no uuid is provided, we return { cloud: undefined, ... }.
~Should we return an empty additionalContext {} if the uuid is not provided or if the hit is empty?~

Nevermind, it seems `alert.setContext()` filters the undefined values out.